tests/platform_tests/daemon/test_thermalctld.py

```diff
@@ -0,0 +1,208 @@
+    
+    data_after_restart = wait_data(duthost)
+    pytest_assert(data_after_restart == data_before_restart, 'DB data present before and after restart does not match')
+
+
+def test_pmon_thermalctld_term_and_start_status(check_daemon_status, duthosts, rand_one_dut_hostname, data_before_restart):
+    """
+    @summary: This test case is to check the thermalctld terminated and restarted status
+    """
+    duthost = duthosts[rand_one_dut_hostname]
+
+    skip_release(duthost, ["201811", "201911"])
+
+    pre_daemon_status, pre_daemon_pid = duthost.get_pmon_daemon_status(daemon_name)
+    logger.info("{} daemon is {} with pid {}".format(daemon_name, pre_daemon_status, pre_daemon_pid))
+
+    duthost.stop_pmon_daemon(daemon_name, SIG_TERM, pre_daemon_pid)
+
+    wait_until(50, 10, 0, check_expected_daemon_status, duthost, expected_running_status)
+
+    post_daemon_status, post_daemon_pid = duthost.get_pmon_daemon_status(daemon_name)
+    pytest_assert(post_daemon_status == expected_running_status,
+                          "{} expected restarted status is {} but is {}".format(daemon_name, expected_running_status, post_daemon_status))
+    pytest_assert(post_daemon_pid != -1,
+                          "{} expected pid is -1 but is {}".format(daemon_name, post_daemon_pid))
+    pytest_assert(post_daemon_pid > pre_daemon_pid,
+                          "Restarted {} pid should be bigger than {} but it is {}".format(daemon_name, pre_daemon_pid, post_daemon_pid))
+    data_after_restart = wait_data(duthost)
+    pytest_assert(data_after_restart == data_before_restart, 'DB data present before and after restart does not match')
+
+
+def test_pmon_thermalctld_kill_and_start_status(check_daemon_status, duthosts, rand_one_dut_hostname, data_before_restart):
+    """
+    @summary: This test case is to check the thermalctld killed unexpectedly (automatically restarted) status
+    """
+    duthost = duthosts[rand_one_dut_hostname]
+    pre_daemon_status, pre_daemon_pid = duthost.get_pmon_daemon_status(daemon_name)
+    logger.info("{} daemon is {} with pid {}".format(daemon_name, pre_daemon_status, pre_daemon_pid))
+
+    duthost.stop_pmon_daemon(daemon_name, SIG_KILL, pre_daemon_pid)
+
+    daemon_status, daemon_pid = duthost.get_pmon_daemon_status(daemon_name)
+    pytest_assert(daemon_status != expected_running_status,
+                          "{} unexpected killed status is not {}".format(daemon_name, daemon_status))
+
+    time.sleep(10)
+
+    post_daemon_status, post_daemon_pid = duthost.get_pmon_daemon_status(daemon_name)
+    pytest_assert(post_daemon_status == expected_running_status,
+                          "{} expected restarted status is {} but is {}".format(daemon_name, expected_running_status, post_daemon_status))
+    pytest_assert(post_daemon_pid != -1,
+                          "{} expected pid is -1 but is {}".format(daemon_name, post_daemon_pid))
+    pytest_assert(post_daemon_pid > pre_daemon_pid,
```

Comment:
post_daemon_pid != pre_daemon_pid

Reply:
I am expecting that the PMON doesn't gets restarted because of this single daemon restarting. If the PMON docker doesn't get restarted then the pid will increase.